amethyst_ui/src/text.rs

```diff
@@ -139,503 +134,132 @@ impl Component for TextEditing {
-            *t = transform
-                .get(entity)
-                .expect("Unreachable: Entities are collected from a prepopulated cache")
-                .tab_order;
-        }
-
-        // Attempt to insert the new entities in sorted position.  Should reduce work during
-        // the sorting step.
-        let transform_set = transform.mask().clone();
-        {
-            // Create a bitset containing only the new indices.
-            let new = (&transform_set ^ &self.tab_order_cache.cached) & &transform_set;
-            for (entity, transform, _new) in (&*entities, &transform, &new).join() {
-                let pos = self
-                    .tab_order_cache
-                    .cache
-                    .iter()
-                    .position(|&(cached_t, _)| transform.tab_order < cached_t);
-                match pos {
-                    Some(pos) => self
-                        .tab_order_cache
-                        .cache
-                        .insert(pos, (transform.tab_order, entity)),
-                    None => self
-                        .tab_order_cache
-                        .cache
-                        .push((transform.tab_order, entity)),
-                }
-            }
-        }
-        self.tab_order_cache.cached = transform_set;
-
-        // Sort from smallest tab order to largest tab order, then by entity creation time.
-        // Most of the time this shouldn't do anything but you still need it for if the tab orders
-        // change.
-        self.tab_order_cache
-            .cache
-            .sort_unstable_by(|&(t1, ref e1), &(t2, ref e2)| {
-                let ret = t1.cmp(&t2);
-                if ret == Ordering::Equal {
-                    return e1.cmp(e2);
-                }
-                ret
-            });
-
-        for text in (&mut text).join() {
+        // Normalize text to ensure we can properly count the characters.
+        // TODO: Possible improvement to be made if this can be moved only when inserting characters into ui text.
+        for text in (&mut texts).join() {
             if (*text.text).chars().any(is_combining_mark) {
                 let normalized = text.text.nfd().collect::<String>();
                 text.text = normalized;
             }
         }
 
+        // TODO: Finish TextEditingCursorSystem and remove this
         {
-            let mut focused_text_edit = focused
-                .entity
-                .and_then(|entity| zip_options(text.get_mut(entity), editable.get_mut(entity)));
-            if let Some((ref mut _focused_text, ref mut focused_edit)) = focused_text_edit {
-                focused_edit.cursor_blink_timer += time.delta_real_seconds();
-                if focused_edit.cursor_blink_timer >= 1.0 / CURSOR_BLINK_RATE {
-                    focused_edit.cursor_blink_timer = 0.0;
+            if let Some((text_editing, _)) = (&mut text_editings, &selecteds).join().next() {
```

Comment:
Ah I was wondering how we were going to handle this.  Maybe somewhere in the docs we should make it clear that you can't edit multiple text fields simultaneously.

Reply:
If you are crazy I assume you could. Its not impossible either, sublime text does have the concept of multiple cursors.

Reply:
Yeah but that's not a common need for videogames so I'd say shelve that enhancement and just point out that it's not there.

Reply:
I don't understand why you don't want this to be allowed? To select multiple things at once, the developer has to opt-in the multi select group for the text fields, so its not like the users will be able to select both at once if they are not supposed to.

Reply:
Sorry I must not be making myself very clear, I just want some comments to say "we don't support editing multiple text fields right now."

Reply:
But in theory we do :thinking:

Reply:
This will only retrieve and process events for the first selected text field it finds.  That's what `.join().next()` does.

Reply:
oops will fix asap :3